### ISSUE TITLE
How to disable and re-enable indexing for a collection?

### ISSUE THREAD

**Issue by AdirthaBorgohain** [qdrant/qdrant_client#109]
I have a lot of vectors (around 5 million) that I need to upload and since I will be uploading them all in one go, I was thinking disabling the index would make the process faster (according to the docs).

However, I am a bit confused as to how to use the python client to disable the index. In the official [docs](https://qdrant.tech/documentation/collections/?selector=aHRtbCA%2BIGJvZHkgPiBkaXY6bnRoLW9mLXR5cGUoMSkgPiBzZWN0aW9uOm50aC1vZi10eXBlKDIpID4gZGl2ID4gZGl2ID4gZGl2ID4gYXJ0aWNsZSA%2BIHA6bnRoLW9mLXR5cGUoMTUp) , as an example for python, the following is given:

```
client.update_collection(
    collection_name="{collection_name}",
    optimizer_config=models.OptimizersConfigDiff(
        max_segment_size=10000
    )
)
```
Would the value of `max_segment_size` need to be set to 0 inorder to disable the index? Also, weirdly for the example using http, instead of `max_segment_size`, `indexing_threshold` is set to 10000 in the same example. So what's the correct way to disable the index?

**Comment by monatis** [comment]
Hi @AdirthaBorgohain, you may set `indexing_threshold` to a very large number before uploading to disable indexing temporarily. You can re-enable indexing by setting it to a reasonable value. The [example here in docs](https://qdrant.tech/documentation/collections/?selector=aHRtbCA%2BIGJvZHkgPiBkaXY6bnRoLW9mLXR5cGUoMSkgPiBzZWN0aW9uOm50aC1vZi10eXBlKDIpID4gZGl2ID4gZGl2ID4gZGl2ID4gYXJ0aWNsZSA%2BIHA6bnRoLW9mLXR5cGUoMTUp#update-collection-parameters) demonstrates enabling indexing by setting `indexing_threshold` to 10000.Hi @AdirthaBorgohain, you may set `indexing_threshold` to a very large number before uploading to disable indexing temporarily. You can re-enable indexing by setting it to a reasonable value. The [example here in docs](https://qdrant.tech/documentation/collections/?selector=aHRtbCA%2BIGJvZHkgPiBkaXY6bnRoLW9mLXR5cGUoMSkgPiBzZWN0aW9uOm50aC1vZi10eXBlKDIpID4gZGl2ID4gZGl2ID4gZGl2ID4gYXJ0aWNsZSA%2BIHA6bnRoLW9mLXR5cGUoMTUp#update-collection-parameters) demonstrates enabling indexing by setting `indexing_threshold` to 10000.

**Comment by AdirthaBorgohain** [comment]
@monatis Understood! Thanks a lot for the help. Will do that. However, do you think the python example in the docs needs to be fixed since it is setting `max_segment_size` rather than setting `indexing_threshold` which the http example correctly uses?

**Comment by monatis** [comment]
Yes, definitely. Already raised a PR for the fix qdrant/docs#52

Thanks for reporting!

**Comment by mathpal-minakshi** [comment]
Hi @monatis It seems ***indexing_threshold=0*** is not working for me. 
![image](https://github.com/user-attachments/assets/3ca377c6-f83c-4183-a308-b04bb94ff362)
![image](https://github.com/user-attachments/assets/722589df-2bd6-417a-8f02-c962555bd384)

AS evident with the above snip, ***indexing threshold is 0***. It's a maze for me now. How do I handle this. It's just a small sample that I have uploaded. My actual data is taking almost 4-5 hours this way ***(indexing while uploading)***.

*** following is the code that I am using **

![image](https://github.com/user-attachments/assets/0d709647-0531-4a66-8fed-440c6e334bfd)
![image](https://github.com/user-attachments/assets/faa61b71-550c-4379-9036-899f205492cd)
